src/app/courses/courses.component.ts

```diff
@@ -241,6 +251,13 @@ export class CoursesComponent implements OnInit, AfterViewInit, OnDestroy {
     this.selectedNotEnrolled = selected.reduce((count, id) => {
       return this.hasSteps(id) ? count + (this.userShelf.courseIds.indexOf(id) === -1 ? 1 : 0) : count;
     }, 0);
+
+  }
+
+  countSelectNotLeft(selected: any) {
```

Comment:
This function is nearly identical to `countSelectNotEnrolled`.  Since a course is either enrolled or not, the count of the courses enrolled is <# selected> - <# not enrolled> and vice versa.  Rather than duplicating code, we could just calculate it that way.